travis.sh

```diff
@@ -33,6 +33,16 @@ function run_script() {
 function run_docker() {
    run_script BEFORE_DOCKER_SCRIPT
 
+    # get ci enviroment parameters to pass into docker for codecov
+    # this is a list of docker parameters to include enviroment variables
+    export CI_ENV_PARAMS=`bash <(curl -s https://codecov.io/env)`
+    # these two varaibles are needed for codecov to work locally
+    set +u  # stop variable checking
+    # test if they are set before setting as travis sets them
+    [[ -z ${VCS_BRANCH_NAME} ]] && export VCS_BRANCH_NAME=$(git rev-parse --abbrev-ref HEAD)
```

Comment:
This is needed for local (non-Travis) usage, I guess?

Reply:
yes

Reply:
Could you please get rid off `set +u` / `set -u` by using corresponding defaults?

Reply:
What do you mean by corresponding defaults, is that a different way of testing if a variable is set?

Reply:
I implemented my suggestion in https://github.com/PickNikRobotics/moveit_ci/pull/1.
Having this feature added, you should also augment the unit tests accordingly.

Reply:
I added unit tests for this.